resources/assets/js/components/JobDetails/JobDetailsPage.tsx

```diff
@@ -0,0 +1,145 @@
+import { Job } from "../../models/types";
+import React, { useEffect } from "react";
+import { JobDetailsIntl } from "./JobDetails";
+import { RootState } from "../../store/store";
+import { DispatchType } from "../../configureStore";
+import {
+  fetchJob,
+  createJob,
+  updateJob,
+  setSelectedJob,
+} from "../../store/Job/jobActions";
+import {
+  getJob as selectJob,
+  getSelectedJob,
+} from "../../store/Job/jobSelector";
+import ReactDOM from "react-dom";
+import { connect } from "react-redux";
+import RootContainer from "../RootContainer";
+import ProgressTracker from "../ProgressTracker/ProgressTracker";
+import { ProgressTrackerItem } from "../ProgressTracker/types";
+
+interface JobDetailsPageProps {
+  jobId: number | null;
+  job: Job | null;
+  loadJob: (jobId: number) => void;
+  handleCreateJob: (newJob: Job) => Promise<boolean>;
+  handleUpdateJob: (newJob: Job) => Promise<boolean>;
+}
+
+const JobDetailsPage: React.FunctionComponent<JobDetailsPageProps> = ({
+  jobId,
+  job,
+  loadJob,
+  handleCreateJob,
+  handleUpdateJob,
+}): React.ReactElement => {
+  useEffect((): void => {
+    if (jobId) {
+      loadJob(jobId);
+    }
+  }, [jobId, loadJob]);
+
+  const waitingForJob = jobId !== null && job === null;
+  const handleModalCancel = (): void => {};
+  const handleModalConfirm = (): void => {
+    window.location.href = "/manager/jobs";
+  }; // TODO: go to next page
+  const handleSubmit = job ? handleUpdateJob : handleCreateJob;
+  const progressTrackerItems: ProgressTrackerItem[] = [
+    { state: "active", label: "Step 01", title: "Job Info" },
+    { state: "complete", label: "Step 02", title: "Work Env." },
+    { state: "error", label: "Step 03", title: "Impact" },
+  ];
+  return (
+    <section>
+      <ProgressTracker
+        items={progressTrackerItems}
+        backgroundColor="black"
+        fontColor="white"
+        classNames="manager-jpb-tracker"
+        itemsWrapperClassNames="tracker manager-jpb-tracker-wrapper"
+      />
+      {waitingForJob ? (
+        <div
+          data-c-container="form"
+          data-c-padding="top(triple) bottom(triple)"
+        >
+          <h3
+            data-c-font-size="h3"
+            data-c-font-weight="bold"
+            data-c-margin="bottom(double)"
+          >
+            Job Loading...
```

Comment:
We'll need to localize this

Reply:
It needs more ui love than just that. But localizing is easy.